python/cuml/manifold/t_sne.pyx

```diff
@@ -163,6 +163,13 @@ class TSNE(Base):
     you run TSNE a few times to settle on the best configuration. Notice
     specifying random_state and fixing it across runs can help, but TSNE does
     not guarantee similar results each time.
+
+    Reference Implementation
+    -------------------------
+    The CUDA implementation is derived from the excellent CannyLabs open source
+    implementation here: https://github.com/CannyLab/tsne-cuda/. The CannyLabs
+    code is licensed according to the conditions in cuml/cpp/src/tsne/
+    cannylabs_tsne_license.txt. We extend our big thanks to CannyLabs!
```

Comment:
Looks great. I'd remove the final bit "We extend our big thanks to CannyLabs!" but add the lines, "A full description of their approach is available in their article t-SNE-CUDA: GPU-Accelerated t-SNE and its Applications to Modern Data (https://arxiv.org/abs/1807.11824)."

Always nice to include a pointer to the source so they can get citations!